python/ray/client.py

```diff
@@ -0,0 +1,97 @@
+import subprocess
+
+# TODO (dsuo): eventually remove this dependency
+import requests
+import pyarrow
+# import pickle
+# import numpy as np
+
+# TODO (dsuo): rename so it's clear what "client" means
+class Client(object):
+    """A class used to proxy communication between a Ray client
+    and a remote Ray head node.
+    """
+
+    def __init__(self,
+                 gateway_address,
+                 gateway_port=5432,
+                 gateway_data_port=5000,
+                 client_socket_name=None):
+        """Initialize a new client.
+
+        Args:
+            gateway_address (str): The IP address of the head node / gateway.
+            gateway_port (int): The gateway's port.
+            data_port (int): The gateway's port for transferring data.
+            client_socket_name (str): The named pipe that forwards local
+                scheduler client to remote head node for processing.
+
+        Returns:
+            A new Client object
+        """
+        self.gateway_address = gateway_address
+        self.gateway_port = gateway_port
+        self.gateway_data_port = gateway_data_port
+        self.client_socket_name = client_socket_name
+        self.serialization_context = None
+        self.url = "http://{}:{}".format(
+            self.gateway_address,
+            self.gateway_data_port)
+        
+        if client_socket_name is not None:
```

Comment:
Is there a one-to-one mapping between `Client` objects and named pipes for forwarding  info? If so, should we just create the named pipe in the constructor here?